### DISCUSSION TITLE
How to include a modulus operation in CasADi function

### DISCUSSION THREAD

**Discussion by billtubbs** [casadi/casadi#3735] [Q&A]
I thought this would be a trivial question so I posted it on stackoverflow but so far, no satisfactory answer:

- [How to include a modulus operation in CasADi function](https://stackoverflow.com/questions/78625360/how-to-include-a-modulus-operation-in-casadi-function)

Any advice appreciated (either here on the stackoverflow post).

**Comment by jgillis** [comment]
Maybe you can find something useful here: https://github.com/casadi/casadi/issues/2767

**Reply by billtubbs** [reply]
Thanks, from what I could see there, the answer is no - i.e. there is no equivalent to Python % or numpy.mod.  Am I right?

**Reply by billtubbs** [reply]
Thinking about it a bit more, in my specific application (i.e. angles) maybe I should avoid any kind of `mod` and use `arctan2(sin(theta), cos(theta))` or just avoid angles completely and work with (x,y) vectors.

It introduces an additional state variable but might be a more robust approach.  Is this common practice?  E.g. in robotics.  Or how do people deal with angle conversions?
